### PR TITLE
bsearch_custom arguments for godot 4.0

### PR DESCRIPTION
**Pull request by hungrymonkey**
name - type - description
func - callable - compare function object
before - boolean - first and second resolver

https://github.com/godotengine/godot-docs/issues/4564

My bad for closing my last pr. I forgot to add a code block and `doc/tools/makerst.py doc/classes/Array.xml` didn't catch it

### CODE REVIEW

File: doc/classes/Array.xml [hunk 1]
```diff
@@ -197,6 +197,25 @@
...
+				func cardinal_to_algebraic(a):
+				    match a:
+				        "one":
+				            return 1
+				        "two":
+				            return 2
+				        "three":
+				            return 3
+				        "four":
+				            return 4
```
Comment (akien-mga) [inline doc/classes/Array.xml:210]:
```suggestion
				        "four":
				            return 4
				        _:
				            return 0
```
To match the `3.2` version (1716423c9797c0f8c368e2bc8ace018db384f66d).

File: doc/classes/Array.xml [hunk 2]
```diff
@@ -197,6 +197,25 @@
...
+				
+				func compare(a,b):
```
Comment (akien-mga) [inline doc/classes/Array.xml:212]: I tried to get you to change this with 3 suggestions so far... so let's try a more detailed approach.

The Godot class reference follows the GDScript style guide, and this style guide requires a space after a comma: https://docs.godotengine.org/en/stable/getting_started/scripting/gdscript/gdscript_styleguide.html#whitespace

File: doc/classes/Array.xml [hunk 3]
```diff
@@ -197,6 +197,25 @@
...
+				    return cardinal_to_algebraic(a) &lt; cardinal_to_algebraic(b)
+				
+				func _ready():
+				    var a = ["one", "two", "three", "four"]
+				    print(a.bsearch_custom("three", compare, true)) #Expected value is 2
```
Comment (akien-mga) [inline doc/classes/Array.xml:217]: Again:
```suggestion
				    print(a.bsearch_custom("three", compare, true)) # Expected value is 2.
```
As per the GDScript style guide: https://docs.godotengine.org/en/stable/getting_started/scripting/gdscript/gdscript_styleguide.html#comment-spacing
(Do note the trailing dot too.)

File: doc/classes/Array.xml (hunk 2 above)
Comment (hungrymonkey) [reply doc/classes/Array.xml:214]: I added your suggestions. Somehow my git repo is messed up. Maybe I am tired.

**Comment by hungrymonkey** [comment]
damn it, it commited twice

**Comment by hungrymonkey** [comment]
I will reopen it tomorrow. I seem to be making too many careless errors today

**Pull request closed without merging**